static/images/diagrams/image_workflow.txt

```diff
@@ -0,0 +1,42 @@
+# Use http://sequencediagram.org/ to generate image
+
+title Image-based provisioning with Puppet
+
+participant User
+participant "Provisioned Host" as Host
+participant "Foreman" as Foreman
+participant "Foreman Proxy" as Proxy
+participant "Hypervisor" as HVM
+participant DHCP
+participant DNS
+participant "Puppet master" as Puppet
+
+linear
+autonumber
+
+note over Foreman:has an Image with credentials defined
+User->Foreman:clicks on New Host
+Foreman->Proxy:requests free IP address
+User->Foreman:selects virtualization resource and submits
+Foreman->HVM:creates new VM
+HCM->Foreman:reports MAC address
+Proxy->DHCP:forwards free IP request
+Foreman->Proxy:creates DHCP reservation
+Proxy->DHCP:forwards DHCP reservation
+Foreman->Proxy:creates DNS records
+Proxy->DNS:forwards DNS records
+Foreman->Proxy:creates DNS records
+Foreman->HVM:starts the VM
+Host->DHCP:requests the reserved IP
+note over Host:boots up
+Foreman->Host:executes finish script via ssh
+
+== Finish script ==
```

Comment:
This section should be the same as cloud_ssh_workflow, where the SSH script signs the Puppet cert + the build is marked as finished, then the stuff below about Puppet runs is in the first Puppet run section.